cmd/api/src/api/v2/auth/oidc.go

```diff
@@ -254,15 +256,17 @@ func getOIDCClaims(reqCtx context.Context, provider *oidc.Provider, ssoProvider
 }
 
 func jitOIDCUserCreation(ctx context.Context, ssoProvider model.SSOProvider, claims oidcClaims, u jitUserCreator) error {
-	if role, err := u.GetRole(ctx, ssoProvider.Config.AutoProvision.DefaultRoleId); err != nil {
-		return fmt.Errorf("get role: %v", err)
+	if roles, err := SanitizeAndGetRoles(ctx, ssoProvider.Config.AutoProvision, claims.Roles, u); err != nil {
+		return fmt.Errorf("sanitize roles: %v", err)
+	} else if len(roles) != 1 {
+		return fmt.Errorf("invalid roles %v", roles.Names())
```

Comment:
If there are 0 roles, you'll get an error that ends with `invalid roles: <nil>` which isn't super nice. Should we change this to a  two step check, once to see if it's less than 1 and once for greater than 1? We could give better error messages to help us later on